### DISCUSSION TITLE
what‘s the difference between ’Streaming Compression Functions‘ in lz4.h and functions in lz4frame.h？

### DISCUSSION THREAD

**Discussion by xiaodou1117** [lz4/lz4#1503] [Q&A] [answered]
what‘s the difference between ’Streaming Compression Functions‘ in lz4.h and functions in lz4frame.h？
I mean，why ’Streaming Compression Functions‘ are put in lz4.h not lz4frame.h？
There are frame compression functions in both lz4.h and lz4frame.h，whats their differences and why put functions like that？

**Comment by t-mat** [comment, marked as answer]
- `lz4.h` provides "low level" functions.  It doesn't contain any interoperability layer.  You must care about everything.
- `lz4frame.h` provides "relatively higher level" functions.  It conforms the LZ4 frame format and provides interoperability between platforms.   Therefore, you don't need to care about many things (OS / CPU / programming language / client / server, etc) with it.

If you don't have any idea about them, we highly recommend to use `lz4frame.h` first.

**Reply by xiaodou1117** [reply]
thank you very much. it helps a lot
